### PR TITLE
FS-3184: Add "All uploaded documents" functionality

### PR DESCRIPTION
**Pull request by tferns**
### Change description

- Add route to get theme answers dynamically for all file upload field types

---

- [x] Unit tests and other appropriate tests added or updated
- [x] README and other documentation has been updated / added (if needed)
- [x] Commit messages are meaningful and follow good commit message guidelines

### How to test

- Check out this branch
- Call the endpoint via http://localhost:3005/ and an application id
- Observe the JSON response

### Screenshots of UI changes (if applicable)

image

### Related 

- https://github.com/communitiesuk/funding-service-design-assessment/pull/319
- https://github.com/communitiesuk/funding-service-design-fund-store/pull/100

### CODE REVIEW

File: api/routes/subcriterias/get_sub_criteria.py [hunk 1]
```diff
@@ -254,15 +269,25 @@ def map_single_field_answer(theme: list, questions: dict) -> str:
...
+    application_id: str, theme_id: str, themes_fields: list[dict]
 ):
     """function maps answers from application with assessor task list
     themes through field ids.
     Args: application_id, theme_id, fund_id, round_id
     Exceptions: returning custom exception along with openapi
     validation detail.
     """
-
-    themes_fields = get_themes_fields(theme_id, fund_id, round_id)
```
Comment (gio-karunakaran) [inline api/routes/subcriterias/get_sub_criteria.py:265]: how comes we're removing themes_fields from this func?

File: api/routes/subcriterias/get_sub_criteria.py [hunk 2]
```diff
@@ -74,6 +74,21 @@ def get_themes_fields(
         return f"Incorrect theme id -> {theme_id}"
 
 
+def get_all_uploaded_document_field_answers(
```
Comment (gio-karunakaran) [inline api/routes/subcriterias/get_sub_criteria.py:77]: can we filter these answers with application_id also? i.e. combine the function here with the 'get_all_uploaded_document_theme_answers()' function?

**Review by gio-karunakaran — COMMENTED** [review]
Just a couple of Qs!

File: api/routes/subcriterias/get_sub_criteria.py (hunk 1 above)
Comment (tferns) [reply api/routes/subcriterias/get_sub_criteria.py:265]: Hey!  The signature of the method has changed, so we pass `themes_fields` in, was split into another function so we could re-use this functionality with the `all_uploaded_documents` functionality!

File: api/routes/subcriterias/get_sub_criteria.py (hunk 2 above)
Comment (tferns) [reply api/routes/subcriterias/get_sub_criteria.py:77]: Hey - not sure what you mean by filter by application id? I think at this point we only have data specific to a single application!

Comment (gio-karunakaran) [reply api/routes/subcriterias/get_sub_criteria.py:77]: perfect!

**Review by gio-karunakaran — APPROVED** [review]
👌🏾

**Pull request merged**